genie-web/src/main/java/com/netflix/genie/web/services/impl/AgentRoutingServiceImpl.java

```diff
@@ -87,5 +104,63 @@ public void handleClientConnected(@NotBlank final String jobId) {
     public void handleClientDisconnected(@NotBlank final String jobId) {
         log.info("Agent executing job {} disconnected", jobId);
         this.agentConnectionPersistenceService.removeAgentConnection(jobId, genieHostInfo.getHostname());
+
+        for (AgentConnectionObserver agentConnectionObserver : agentConnectionObservers) {
+            this.connectionObserversNotificationExecutor.execute(
```

Comment:
This could throw (ah, the beauty of unchecked exceptions!).
I think the exception can propagate all the way back to a gRPC thread, which might permanently disable or block the server.